File: Baekjoon/Class2/29.9012 괄호.py
import sys
input = sys.stdin.readline
write = sys.stdout.write

def vps(ps): #valid parenthesis string
    count = 0
    for ch in ps:
        if ch == "(": #왼쪽 괄호 count += 1
            count += 1
        else: #오른쪽 괄호 count -= 1
            if count == 0: return False #count 빔
            else: count -= 1
    return count == 0 #남은 괄호 없으면 성공

def main():
    t = int(input())
    for _ in range(t):
        ps = input().rstrip()
        if vps(ps): write("YES\n")
        else: write("NO\n")

main()

File: Baekjoon/Class2/29.9012 괄호.js
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").split("\n");

const vps = (ps) => { // valid parenthesis string
    let count = 0;
    for(const ch of ps)
        if(ch === "(") // 왼쪽 괄호 count++
            count++;
        else // 오른쪽 괄호 count--
            if(count === 0) return false; // count 빔
            else count--;
    return count === 0; // 남은 괄호 없으면 성공
}

const t = Number(input[0]);
const out = [];
for(let i = 1; i <= t; i++)
    out.push(vps(input[i].trim()) ? "YES" : "NO");
console.log(out.join("\n"));